web/analytics/models.py

```
@@ -0,0 +1,109 @@
+    updated_at = models.DateTimeField(auto_now=True)
+    
+    class Meta:
+        unique_together = ['student', 'course']
+        verbose_name = "Student Analytics"
+        verbose_name_plural = "Student Analytics"
+    
+    def __str__(self):
+        return f"{self.student.username}'s analytics for {self.course.title}"
+    
+    @property
+    def completion_percentage(self):
```

Ruff (0.8.2)

Missing return type annotation for public function `completion_percentage`

(ANN201)